camelot/view/qml_view.py

```diff
@@ -108,9 +107,15 @@ def register(self, gui_context, model=None):
         self.gui_contexts[context_id] = gui_context
         if model is not None:
             self.models[context_id] = model
+            model.destroyed.connect(self.remove_model)
         gui_context.context_id = context_id
         return context_id
 
+    def remove_model(self):
```

Comment:
je kan deze method best expliciet als slot declareren, dan is er een check bij het connecten op signature etc